### PR TITLE
Add LEAPP_DEVEL_DM_DISABLE_UDEV env var

### PR DESCRIPTION
**Pull request by Rezney**
When, during dnf upgrade transaction, some package calls
device_mapper which requires udev to properly release a cookie
set by device mapper, the transaction gets stuck as udev on the
host cannot see namespaced IPC inside our container.

We workaround this by setting "DM_DISABLE_UDEV=1" env var.

### CODE REVIEW

**Comment by Rezney** [comment]
Need to find out a way to pass the env var into initramdisk...

**Comment by pirat89** [comment]
Please, update commit message and description. You skipped the LEAPP_DEVEL_ prefix for the var name :)

**Comment by Rezney** [comment]
> Please, update commit message and description. You skipped the LEAPP_DEVEL_ prefix for the var name :)

I wanted to mention the original env var we are using, but you are right I should mention both. Anyway it is not working atm...

**Comment by Rezney** [comment]
@vinzenz yep, already relalized that :) We were thinking about using the soon-to-be-added configuration actor for this. Or do you have other idea?

**Comment by Rezney** [comment]
@vinzenz could you please take a look at this when you have some free cycles? thx

File: repos/system_upgrade/el7toel8/libraries/dnfplugin.py [hunk 1]
```diff
@@ -106,7 +106,8 @@ def _transaction(context, stage, target_repoids, tasks, test=False):
                     stage,
                     DNF_PLUGIN_DATA_PATH
                 ],
-                callback_raw=utils.logging_handler
+                callback_raw=utils.logging_handler,
+                env=cmd_env_vars or {}
```
Comment (vinzenz) [inline repos/system_upgrade/el7toel8/libraries/dnfplugin.py]: IIRC the or is irrelevant None is fine
I think that’s the default value on run

File: repos/system_upgrade/el7toel8/libraries/config/__init__.py [hunk 2]
```diff
@@ -0,0 +1,10 @@
+from leapp.libraries.stdlib import api
+
+
+def get_env_var_value(name):
```
Comment (vinzenz) [inline repos/system_upgrade/el7toel8/libraries/config/__init__.py]: Why not just get_env with an or value and default value None as fallback?

File: repos/system_upgrade/el7toel8/libraries/dnfplugin.py [hunk 3]
```diff
@@ -89,7 +89,7 @@ def backup_debug_data(context):
             api.current_logger().warn('Failed to copy debugdata. Message: {}'.format(e.message), exc_info=True)
 
 
-def _transaction(context, stage, target_repoids, tasks, test=False):
+def _transaction(context, stage, target_repoids, tasks, cmd_env_vars=None, test=False):
```
Comment (vinzenz) [inline repos/system_upgrade/el7toel8/libraries/dnfplugin.py]: Why not just env or at least cmd_env

Comment (Rezney) [reply repos/system_upgrade/el7toel8/libraries/dnfplugin.py]: Fixed the namings. It was indeed too long.

File: repos/system_upgrade/el7toel8/libraries/config/__init__.py (hunk 2 above)
Comment (Rezney) [reply repos/system_upgrade/el7toel8/libraries/config/__init__.py]: like this?

File: repos/system_upgrade/el7toel8/libraries/config/__init__.py [hunk 4]
```diff
@@ -0,0 +1,10 @@
+from leapp.libraries.stdlib import api
+
+
+def get_env(name, default=None):
+    """ Return Leapp environment variable value if matched by name """
+    env_vars = api.current_actor().configuration.leapp_env_vars
```
Comment (vinzenz) [inline repos/system_upgrade/el7toel8/libraries/config/__init__.py]: I think this temporary isn't necessary

File: repos/system_upgrade/el7toel8/libraries/config/__init__.py [hunk 5]
```diff
@@ -0,0 +1,10 @@
...
+    for var in env_vars:
```
Comment (vinzenz) [inline repos/system_upgrade/el7toel8/libraries/config/__init__.py]:
```suggestion
    for var in api.current_actor().configuration.leapp_env_vars:
```

Comment (Rezney) [reply repos/system_upgrade/el7toel8/libraries/config/__init__.py]: why not, done...

**Comment by Rezney** [comment]
rebased

**Comment by Rezney** [comment]
@jmikovic prosim, prosim

**Comment by Rezney** [comment]
@jmikovic so you can leave this for the moment. It may not be working as expected as for now...

File: repos/system_upgrade/el7toel8/libraries/dnfplugin.py [hunk 6]
```diff
@@ -106,7 +106,8 @@ def _transaction(context, stage, target_repoids, tasks, test=False):
                     stage,
                     DNF_PLUGIN_DATA_PATH
                 ],
-                callback_raw=utils.logging_handler
+                callback_raw=utils.logging_handler,
+                env=cmd_env
```
Comment (vinzenz) [inline repos/system_upgrade/el7toel8/libraries/dnfplugin.py]: The problem why this doesn't work is that this is a container execution and environment variables aren't forwarded. The solution to this is that we add additional support for it to the mounting.py in the libraries

**Comment by drehak** [comment]
LGTM.

**Pull request merged**